Feature/auth backend

matiboy opened the pull request:
@Jafnee The focus is actually on authentication backend and its tests, not on the views/templates; that was more of a side effect of preparing for it.

matiboy committed:
feat(verification): Display enter pin view using form

- Enter pin view
- Template with expected login view in mind
- For with split input
widget

TODO: check how the value comes out of the split input

matiboy committed:
feat(verification): Create authentication backend from email and pin to user

- Authentication backend
- Tests

Jafnee commented on django_pin_auth/auth_backend.py:
```diff
@@ -0,0 +1,31 @@
+
+from .models import SingleUseToken
+
+class PinBackend(object):
```
Auth backend should also implement a [`get_user`](https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#writing-an-authentication-backend) method

matiboy replied:
See a4a5ca393ffc553202b266bdea790768a119f8f8

Jafnee commented on django_pin_auth/auth_backend.py:
```diff
@@ -0,0 +1,31 @@
+
+class PinBackend(object):
+    """Authentication backend based on pin value."""
+    def authenticate(self, request, email=None, pin=None):
```
What do you think if instead of checking for User and Token in 2 steps

We just grab the token:

```python3
try:
    token = SingleUseToken.objects.select_related('user').get(user__email=email, token=pin)
except SingleUseToken.DoesNotExist:
    return None
```

Jafnee replied:
\+ using the USERNAME_FIELD instead of `user__email`

matiboy replied:
See a4a5ca393ffc553202b266bdea790768a119f8f8

matiboy committed:
refactor(verification): Refactor to query straight for user

- Implement get_user
- Change to make a single query to find the token, using a join to the
  user's email
